src/backend/vulkan/src/lib.rs

```diff
@@ -663,6 +665,9 @@ impl adapter::PhysicalDevice<Backend> for PhysicalDevice {
             })
             .collect::<Vec<_>>();
 
+        let requested_extensions = requested_features & Features::EXTENSIONS_MASK;
```

Comment:
why is this separation needed? what happens if you just use `requsted_features` for everything?

Reply:
I see now that It is not. This was only a confusion on my side of understanding gfx-hal. Sorry, I am still grappling this huge codebase :). I though It was going directly to a Vulkan structure and missed the conversion.